Changing the menu-sidebar width to be auto rather than a set pixel value

kev-kev opened the pull request:
Resolves #3389

### Description

Changing the left menu bar styling to auto width rather than a set pixel value. I saw it was being set manually to fix #2840 . Setting the width of the menu bar to auto fixes this issue and keeps the drop down arrows visible in Firefox when the menu  drop downs are expanded on mobile.

### Type of change

* Bug fix (non-breaking change which fixes an issue)


### How Has This Been Tested?

Visually confirmed this issue and #2840 is fixed in Chrome and Firefox, rspec passes 

### Screenshots

![Screen Shot 2023-03-07 at 10 18 29 AM](https://user-images.githubusercontent.com/32309015/223467569-76f49b4d-aae5-471e-b826-56d3b9ad122d.png)
![Screen Shot 2023-03-07 at 10 18 50 AM](https://user-images.githubusercontent.com/32309015/223467598-223a2a59-78f4-4122-b4ad-4ea5cbea248c.png)

dorner commented:
Nice! @cielf can you take a QA pass at this?

cielf commented:
I think it's @scooter-dangle who has the best handle on the effect.  But if he can't, I'll take a stab at it.

awwaiid commented:
@kev-kev we had some issues with Safari that we just merged a fix for. Could you see if after you resolve/re-apply your change does it work in Safari still?

awwaiid reviewed (changes requested):
@kev-kev we had some issues with Safari that we just merged a fix for. Could you see if after you resolve/re-apply your change does it work in Safari still?

scooter-dangle commented:
The issue worked on here seems to be fixed in iOS Safari and Android Chrome (see recordings). Would be good to see it with the Safari fix @awwaiid mentioned, since we see that issue present in the recording (the menu drop downs not dropping down).

https://user-images.githubusercontent.com/934707/226192510-82889285-bb23-4405-876e-57997be3d3cb.mov

https://user-images.githubusercontent.com/934707/226192529-2a246e1a-8543-4e89-8466-3483c441a4f4.mov

kev-kev commented:
I'm a little lost on this. My change was just setting the sidebar's width to auto, but it looks like those safari changes are setting the width of a bunch of different classes to specific pixel values. I'm not really sure what all of that is being done for, but when I comment out both their and my changes to app/stylesheets/application.scss I don't see #3389 occurring on iOS Safari.

dorner commented:
Can you validate with your fixes that a) the sidebar expands/collapses correctly on Mac OS Safari, and b) the sections expand/collapse on the right?

kev-kev commented:
Confirmed! After commenting out the changes from the changes from the safari fixes (and this PR, since i forgot `width: auto;` is default behavior), it works on MacOS Safari.

![Screen Shot 2023-03-24 at 12 12 10 PM](https://user-images.githubusercontent.com/32309015/227581804-b477031a-c07e-4ef7-a8f3-bac0fa5235e0.png)

dorner commented:
Nice! @kev-kev can you fix the conflicts and then we can merge this sucker :grin:

cielf commented:
I second that motion!  FWIW @kev-kev - this is going to make a big difference to some of our users who are trying to use the app on phones.

kev-kev commented:
Ok, done! I'm still not sure what the changes I'm removing are for, specifically
`body:not(.sidebar-mini-md):not(.sidebar-mini-xs):not(.layout-top-nav) .content-wrapper, body:not(.sidebar-mini-md):not(.sidebar-mini-xs):not(.layout-top-nav) .main-footer, body:not(.sidebar-mini-md):not(.sidebar-mini-xs):not(.layout-top-nav) .main-header {
   margin-left: 280px;
 }`
So I'm just a bit worried that the styling that's touching some part of the app that's going to break with these changes removed... But I went ahead and removed it and fixed the conflict!

dorner commented:
Let's give it a shot!
